webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala

```diff
@@ -0,0 +1,61 @@
+package com.normation.rudder.services.healthcheck
+
+import java.lang.Runtime.getRuntime
+
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.logger.{HealthcheckLoggerPure => logger}
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Critical
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Warning
+import com.normation.rudder.services.healthcheck.HealthcheckResult.Ok
+import zio.UIO
+import zio.ZIO
+import zio.syntax.ToZio
+
+final case class CheckName(value: String)
+
+trait Check {
+  def name: CheckName
+  def run: IOResult[HealthcheckResult]
+}
+
+sealed trait HealthcheckResult { def msg:String }
+object HealthcheckResult {
+  final case class Ok(val msg: String) extends HealthcheckResult
+  final case class Warning(val msg: String) extends HealthcheckResult
+  final case class Critical(val msg: String) extends HealthcheckResult
+}
+
+class HealthcheckService(checks: List[Check]) {
+
+   def runAll: ZIO[Any, Nothing, List[HealthcheckResult]]  = ZIO.foreach(checks) { c =>
+      for {
+        res <- c.run.catchAll { err =>
+                 HealthcheckResult.Critical(
+                   s"A fatal error was encountered when running check '${c.name}': ${err.fullMsg}"
+                 ).succeed
+               }
+        _   <- logHealthcheck(res)
+      } yield res
+    }
+
+  private[this] def logHealthcheck(check: HealthcheckResult): UIO[Unit] = {
+    check match {
+      case c: Critical => logger.error(s"`Healthcheck has detected a critical issue: ${check.msg}")
+      case w: Warning =>  logger.warn(s"Healthcheck check has detected a warning issue: ${check.msg}")
+      case o: Ok =>       logger.debug(s"Healthcheck check has passed: ${check.msg}")
+    }
```

Comment:
I think this can be more succint and precise:
- we already know it's healthcheck because of the name of the logger, 
- we already know the level (it's in the log),
- we don't have the probe name, it should be added

I think it can be just, for all logs: `${name}: ${check.msg}` and just change the level, so:

```
  private[this] def logHealthcheck(name: String, check: HealthcheckResult): UIO[Unit] = {
    val msg = s"${name}: ${check.msg}"
    check match {
      case _: Critical => logger.error(msg)
      case _: Warning  => logger.warn(msg)
      case _: Ok       => logger.debug(msg)
    }
}
```